netshare/generate/denormalize_fields.py

```diff
@@ -67,72 +74,160 @@ def write_to_csv(
     timeseries: List[np.ndarray],
     data_gen_flag: np.ndarray,
     filename: str,
+    config: Config,
 ) -> None:
     """
     This function dumps the given data to the given directory as a csv format.
     `data_gen_flag` is an indicator showing if the time series for this session
     has ended in this time step.
     """
     os.makedirs(csv_folder, exist_ok=True)
-    csv_path = os.path.join(csv_folder, f"data_{filename}_{random.random()}.csv")
+    csv_path = os.path.join(csv_folder, filename)
     # change session key shape to #session * #attributes
-    session_key = np.concatenate(session_key, axis=1)
+    session_key_numpy = np.array(np.concatenate(session_key, axis=1))
     # change timeseries shape to #session * #time_steps * #features
-    timeseries = np.concatenate(timeseries, axis=2)
+    timeseries_numpy = np.array(np.concatenate(timeseries, axis=2))
 
     with open(csv_path, "w") as f:
         writer = csv.writer(f)
         session_titles = _get_fields_names(session_key_fields)
         timeseries_titles = _get_fields_names(timeseries_fields)
+        raw_metadata_field_names = [col.column for col in config["metadata"]]
+        raw_timeseries_filed_names = [col.column for col in config["timeseries"]]
+        session_titles = [
+            f for i, f in enumerate(session_titles) if f in raw_metadata_field_names
+        ]
+        session_titles_idx = [
+            i for i, f in enumerate(session_titles) if f in raw_metadata_field_names
+        ]
+        timeseries_titles = [
+            f
+            for i, f in enumerate(timeseries_titles)
+            if f in raw_timeseries_filed_names
+        ]
+        timeseries_titles_idx = [
+            i
+            for i, f in enumerate(timeseries_titles)
+            if f in raw_timeseries_filed_names
+        ]
+
+        if config["timestamp"]["generation"]:
+            timeseries_titles.append(config["timestamp"]["column"])
+            if config["timestamp"]["encoding"] == "interarrival":
+                # Find `flow_start` and `interarrival_within_flow` index
+                flow_start_idx, interarrival_within_flow_idx = None, None
```

Comment:
I'm not expecting to see this type of data manipulation in a function named `write_to_csv`. Can you please write it in a separated function, with a name that tells us what you're trying to do?